database-dispatcher/database-dispatcher.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const path = require('path');
+
+const DatabaseDispatcherError = require('./database-dispatcher-error');
+
+const DB_DRIVERS = {
+	mysql: '@janiscommerce/mysql',
+	mongodb: '@janiscommerce/mongodb'
+};
+
+class DatabaseDispatcher {
+
+	static get dbTypes() {
+		return Object.keys(DB_DRIVERS);
+	}
+
+	static get configPath() {
+		return path.join(process.cwd(), 'config', 'database.json');
+	}
+
+	get databaseConfig() {
+
+		if(!this.config) {
+			try {
+				/* eslint-disable global-require, import/no-dynamic-require */
+				const config = require(this.constructor.configPath);
+				this.config = config;
+
+			} catch(error) {
+				throw new DatabaseDispatcherError('Config not found', DatabaseDispatcherError.codes.CONFIG_NOT_FOUND);
+			}
+		}
+
+		return this.config;
+	}
+
+	/**
+	 * Evaluates the config object then returns the selected DBDriver
+	 * @param {Object} config database config json
+	 * @returns DBDriver
+	 */
+	static getDBDriver(config) {
+
+		if(config && config.type && this.dbTypes.includes(config.type)) {
+
+			try {
+
+				return require(path.join(process.cwd(), 'node_modules', DB_DRIVERS[config.type]));
```

Comment:
```suggestion
				return require(this.dbTypes[config.type]);
```

Más simple. Para hacer esto es requerido el cambio que sugerí en el _getter_ `dbTypes()`

Reply:
La parte de `path.join(process.cwd(),'node_modules')` es necesaria porque el driver de la DB se instala en el microservicio y no en el package del dispatcher, por ende no lo encuentra si no esta cargado de esta manera.